tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -120,6 +120,43 @@ internal sealed class AsyncToSyncMethodTransformerTests {
 		Assert.AreEqual( "[Blocking] void Bar() { Foo fooFirst = m_foo.Bar( barrer ); }", actual.Value.ToFullString() );
 	}
 
+	[Test]
+	public void CompletedTaskConfigureAwait() {
+		var actual = Transform( @"[GenerateSync] Task BarAsync() { return Task.CompletedTask; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Bar() { return ; }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void FromResult() {
+		var actual = Transform( @"[GenerateSync] Task BarAsync() { return Task.FromResult( baz ); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Bar() { return ( baz ); }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void FromResultGeneric() {
+		var actual = Transform( @"[GenerateSync] Task<Baz> BarAsync() { return Task.FromResult<Baz>( null ); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] Baz Bar() { return ( null ); }", actual.Value.ToFullString() );
+	}
+
+	// Not the best thing to be doing but unlikely this will happen and it should be easily caught
+	[Test]
+	public void DontRemoveUnreturnedFromResult() {
+		var actual = Transform( @"[GenerateSync] Task BarAsync() { var Baz = Task.CompletedTask; return Baz; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Bar() { var Baz = Task.CompletedTask; return Baz; }", actual.Value.ToFullString() );

```

Comment:
This will be a compiler error but that's OK. It would be a bit confusing to diagnose as a user (why is the generator trying to return something here?), so an explicit error would be nicer... even if its opaque, like if we detect we are returning something from a void method then we can just bail and say "uh, can't generate this one sorry".

I'm just thinking out loud. Having a test to demonstrate the current behaviour + a comment is good 👍